File: examples/python/bot_with_dependencies.py
"""
Python bot with external dependencies example.
This bot uses the requests library to make HTTP calls.
"""

import time
import requests
from datetime import datetime

def fetch_quote():
    """Fetch a random quote from an API"""
    try:
        response = requests.get("https://api.quotable.io/random", timeout=5)
        if response.status_code == 200:
            data = response.json()
            return f'"{data["content"]}" - {data["author"]}'
    except Exception as e:
        return f"Failed to fetch quote: {str(e)}"

def main():
    print("=" * 50)
    print("Python Bot with Dependencies")
    print("=" * 50)
    print(f"Started at: {datetime.now().isoformat()}")
    print()
    
    iteration = 0
    
    try:
        while True:
            iteration += 1
            print(f"\n[Iteration {iteration}] {datetime.now().strftime('%H:%M:%S')}")
            
            # Fetch and display a quote
            quote = fetch_quote()
            print(f"Quote of the moment: {quote}")
            
            # Wait before next iteration
            print("Waiting 30 seconds...")
            time.sleep(30)
    
    except KeyboardInterrupt:
        print("\n" + "=" * 50)
        print("Bot stopped")
        print(f"Completed {iteration} iterations")
        print("=" * 50)

if __name__ == "__main__":
    main()

File: examples/nodejs/bot_with_dependencies.js
/**
 * Node.js bot with external dependencies example.
 * This bot uses axios to make HTTP calls.
 */

const axios = require('axios');

const formatTime = () => {
    const now = new Date();
    return now.toTimeString().split(' ')[0];
};

const fetchQuote = async () => {
    try {
        const response = await axios.get('https://api.quotable.io/random', {
            timeout: 5000
        });
        const data = response.data;
        return `"${data.content}" - ${data.author}`;
    } catch (error) {
        return `Failed to fetch quote: ${error.message}`;
    }
};

const main = async () => {
    console.log("=".repeat(50));
    console.log("Node.js Bot with Dependencies");
    console.log("=".repeat(50));
    console.log(`Started at: ${new Date().toISOString()}`);
    console.log();

    let iteration = 0;
    let running = true;

    const runLoop = async () => {
        while (running) {
            iteration++;
            console.log(`\n[Iteration ${iteration}] ${formatTime()}`);

            // Fetch and display a quote
            const quote = await fetchQuote();
            console.log(`Quote of the moment: ${quote}`);

            // Wait before next iteration
            console.log("Waiting 30 seconds...");
            await new Promise(resolve => setTimeout(resolve, 30000));
        }
    };

    // Handle graceful shutdown
    const shutdown = () => {
        running = false;
        console.log("\n" + "=".repeat(50));
        console.log("Bot stopped");
        console.log(`Completed ${iteration} iterations`);
        console.log("=".repeat(50));
        process.exit(0);
    };

    process.on('SIGINT', shutdown);
    process.on('SIGTERM', shutdown);

    await runLoop();
};

main().catch(error => {
    console.error("Fatal error:", error);
    process.exit(1);
});
